Rubberduck.VBEEditor/ComManagement/ComSafeManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+namespace Rubberduck.VBEditor.ComManagement
+{
+    public static class ComSafeManager
+    {
+        private static Lazy<IComSafe> _comSafe = new Lazy<IComSafe>(NewComSafe);
+
+        public static IComSafe GetCurrentComSafe()
+        {
+            return _comSafe.Value;
+        }
+
+        public static void ResetComSafe()
+        {
+            _comSafe = new Lazy<IComSafe>(NewComSafe);
```

Comment:
do we need to ensure that the previous ComSafe has released all references associated with it here?

Reply:
If you mean disposing them, that might be unsafe because thay might still be in use by someone else. In general, the `ComSafe` is only a safety net for the cases where the code owning the com wrapper instances does not handle their disposal.

Reply:
Makes it all the more important to not drop any references in that Safe .... We should possibly pass the references to the new ComSafe created here (and possibly even force creation of that safe)

Reply:
I guess based on the current usage, the method could be renamed to `DisposeAndResetComSafe` and then include disposal of the old safe. This should anyway only ever be called from `Extension` and it only exists to allow reloading RD from the addins menu after unloading it.